page_iterator.go

```diff
@@ -125,20 +170,10 @@ func (pI *PageIterator[T]) GetOdataDeltaLink() *string {
 	return pI.currentPage.oDataDeltaLink
 }
 
-func (pI *PageIterator[T]) next(context context.Context) (PageResult[T], error) {
-	var page PageResult[T]
-
-	resp, err := pI.fetchNextPage(context)
-	if err != nil {
-		return page, err
-	}
-
-	page, err = convertToPage[T](resp)
-	if err != nil {
-		return page, err
-	}
-
-	return page, nil
+// HasNext returns true if there are additional items to iterate.
+func (pI *PageIterator[T]) HasNext() bool {
+	return pI.pauseIndex < len(pI.currentPage.getValue()) ||
+		pI.GetOdataNextLink() != nil && *pI.GetOdataNextLink() != ""
```

Comment:
defensive programing , we can check if the next link is a white space using `strings.TrimSpace(s)`